src/interpreter/library/mod.rs

```diff
@@ -117,37 +121,35 @@ impl StandardLibrarySearcher {
     }
 }
 
-impl LibrarySearcher for StandardLibrarySearcher {
-    fn search_lib(&self, library_name: Located<LibraryName>) -> Option<PathBuf> {
-        match std::env::current_dir() {
-            Ok(pwd) => Some(
-                pwd.join("lib")
-                    .join(library_name.extract_data().path())
-                    .with_extension("sld"),
-            ),
-            Err(_) => None,
-        }
+impl StandardLibrarySearcher {
+    pub fn get_include_library(library_name: &LibraryName) -> Option<&'static str> {
+        let defined_librarys = include::library_strs()
+            .into_iter()
+            .collect::<HashMap<_, _>>();
+        defined_librarys.get(library_name).map(Clone::clone)
     }
 }
 
-#[test]
-fn standard_library_searcher() {
-    let searcher = StandardLibrarySearcher::new();
-    match std::env::current_dir() {
-        Ok(pwd) => {
-            assert_eq!(
-                searcher.search_lib(library_name!("s", 0, "a").into()),
-                Some(
-                    pwd.join("lib")
-                        .join("s")
-                        .join("0")
-                        .join("a")
-                        .with_extension("sld")
-                )
-            );
-        }
-        Err(_) => {
-            assert_eq!(searcher.search_lib(library_name!("s", 0, "a").into()), None);
+impl LibrarySearcher for StandardLibrarySearcher {
+    fn search_lib(
+        &self,
+        library_name: Located<LibraryName>,
+    ) -> Option<Box<dyn Iterator<Item = char>>> {
+        match Self::get_include_library(&library_name) {
+            Some(library_str) => Some(Box::new(library_str.chars())),
+            None => match std::env::current_dir() {
+                Ok(pwd) => {
+                    let path = pwd
+                        .join("lib")
+                        .join(library_name.extract_data().path())
+                        .with_extension("sld");
+                    match file_char_stream(&path) {
+                        Ok(char_stream) => Some(Box::new(char_stream)),
+                        Err(_) => None,
```

Comment:
Error information lost?

Reply:
This is a library searcher, if load char stream from file failed, treat it as a result of library not found, not an error.
Then it is useful to combine this StandardLibrarySearcher with other user defined searcher (implementation of file extension priority, etc).
If the io error information matters, maybe need a log, but not returned by Result<_, _>

Reply:
But you still need to place the "File Not Found" Error in somewhere, and those 
detailed information becomes a single assumption, still lost there.